runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -864,8 +863,10 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
                         .contains(NodeLocator.getLegacyEndpoint(endpoint)))
                 .forEach(endpoint -> {
                     try {
-                        nodeRouterPool.getNodeRouters().get(endpoint).stop();
-                        nodeRouterPool.getNodeRouters().remove(endpoint);
+                        IClientRouter router = nodeRouterPool.getNodeRouters().remove(endpoint);
+                        if (router != null) {
+                            router.stop();
```

Comment:
Just documenting our discussion here.
If the stop fails, there could be a possible leak of the router.
Reference to the router is lost and as experimented, the router does not get garbage collected and the channel remains open.

Reply:
Done.